ee/vellum_ee/workflows/display/nodes/vellum/inline_prompt_node.py

```diff
@@ -41,18 +42,22 @@ def serialize(
         array_display = self.output_display[node.Outputs.results]
         json_display = self.output_display[node.Outputs.json]
         node_blocks = raise_if_descriptor(node.blocks) or []
+
         function_definitions = raise_if_descriptor(node.functions)
 
+        if isinstance(function_definitions, BaseDescriptor):
+            function_definitions = getattr(function_definitions, "instance", [])
+
         ml_model = str(raise_if_descriptor(node.ml_model))
 
         blocks: list = [
             self._generate_prompt_block(block, input_variable_id_by_name, [i]) for i, block in enumerate(node_blocks)
         ]
-        functions = (
-            [self._generate_function_tools(function, i) for i, function in enumerate(function_definitions)]
-            if function_definitions
-            else []
-        )
+        functions = []
+        if function_definitions and isinstance(function_definitions, list):
+            for i, function in enumerate(function_definitions):
+                if callable(function) or isinstance(function, FunctionDefinition):
+                    functions.append(self._generate_function_tools(function, i))
```

Comment:
On second thought, based on my comment below about the new assertions, let's just make sure `functions` here remains empty if we determine that `function_definitions` is a `BaseDescriptor`